halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/provider/v1/OrcaRunner.java

```diff
@@ -0,0 +1,280 @@
+/*
+ * Copyright 2017 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.netflix.spinnaker.halyard.deploy.provider.v1;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.netflix.spinnaker.halyard.core.error.v1.HalException;
+import com.netflix.spinnaker.halyard.core.problem.v1.Problem;
+import com.netflix.spinnaker.halyard.core.problem.v1.ProblemBuilder;
+import com.netflix.spinnaker.halyard.core.tasks.v1.DaemonTaskHandler;
+import com.netflix.spinnaker.halyard.deploy.spinnaker.v1.service.OrcaService.Orca;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import retrofit.RetrofitError;
+
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+@Component
+class OrcaRunner {
+  @Autowired
+  private ObjectMapper objectMapper;
+
+  void monitorTask(Supplier<String> submitTask, Orca orca) {
+    final String id = submitTask.get().substring(1);
+
+    Supplier<Pipeline> getPipeline = () -> {
+      Map<String, Object> execution = orca.getRef(id);
+      return objectMapper.convertValue(execution, Task.class).getExecution();
+    };
+    monitor(getPipeline, orca);
+  }
+
+  void monitorPipeline(Supplier<String> submitPipeline, Orca orca) {
+    final String id = submitPipeline.get().substring(1);
```

Comment:
Good idea, the char is a leading '/' that is returned by orca as a reference to the pipeline.